### DISCUSSION TITLE
is there any document about phone side setup?

### DISCUSSION THREAD

**Discussion by senxiangms** [DeviceFarmer/stf#645] [Q&A] [answered]
I  successfully run stf in virtualbox ubuntu. But when I connect a phone into usb hub, set it to usb debugging mode, but still cannot find it in localhost:7100/#!/devices page. 

any idea?

**Comment by denis99999** [comment, marked as answer]
@senxiangms , did you link your usb device to your virtual machine ?

**Reply by senxiangms** [reply]
After linking usb device to VM, the phone can be managed by stf now. thanks a lot.
